### PR TITLE
TABL: Skip data check for GTTs

### LINKED ISSUE #4074: TABL: Dump while uninstalling GTT
Demo repo
https://gitlab.com/christianguenter/test_gtt

GTT requires strict SQL syntax
https://help.sap.com/doc/abapdocu_750_index_htm/7.50/en-US/abenddic_database_tables_gtt.htm

image
image
image

### PR DESCRIPTION
**Pull request by christianguenter2**
fixes #4074

test repo
https://github.com/abapGit-tests/TABL_GTT.gitr

### CODE REVIEW

**Comment by mbtools** [comment]
We make an assumption here that there is or won't be other types of tables that would raise this exception as well (that are not gtt and users should confirm deletion). 

Standard DD_EXISTS_DATA would also work with cluster and pool tables:

```abap
    SELECT SINGLE tabname tabclass sqltab FROM dd02l INTO ls_dd02l
        WHERE tabname = ms_item-obj_name
        AND as4local = 'A'
        AND as4vers = '0000'.
    IF sy-subrc = 0.
      CALL FUNCTION 'DD_EXISTS_DATA'
        EXPORTING
          reftab          = ls_dd02l-sqltab
          tabclass        = ls_dd02l-tabclass
          tabname         = ls_dd02l-tabname
        IMPORTING
          subrc           = lv_subrc
        EXCEPTIONS
          missing_reftab  = 1
          sql_error       = 2
          buffer_overflow = 3
          unknown_error   = 4
          OTHERS          = 5.
      IF sy-subrc = 0 AND lv_subrc = 0.
        lv_no_ask = abap_false.
      ENDIF.
    ENDIF.
`

**Comment by christianguenter2** [comment]
Yes, that's the far better solution. I updated the PR.

**Review by mbtools — APPROVED** [review]
Nice 👍

File: src/objects/zcl_abapgit_object_tabl.clas.abap [hunk 1]
```diff
@@ -589,27 +590,31 @@ CLASS ZCL_ABAPGIT_OBJECT_TABL IMPLEMENTATION.
...
 
       lv_no_ask = abap_true.
-      SELECT SINGLE tabclass FROM dd02l INTO lv_tabclass
+      SELECT SINGLE tabname tabclass sqltab FROM dd02l
+        INTO CORRESPONDING FIELDS OF ls_dd02l
         WHERE tabname = ms_item-obj_name
         AND as4local = 'A'
         AND as4vers = '0000'.
-      IF sy-subrc = 0 AND lv_tabclass = 'TRANSP'.
+      IF sy-subrc = 0.
```
Comment (larshp) [inline src/objects/zcl_abapgit_object_tabl.clas.abap:598]: if `AND lv_tabclass = 'TRANSP'` we can avoid calling the function module for structures and views?

Comment (mbtools) [reply src/objects/zcl_abapgit_object_tabl.clas.abap:598]: TRANSP or CLUSTER or POOL

Comment (christianguenter2) [reply src/objects/zcl_abapgit_object_tabl.clas.abap]: updated

**Review by mbtools — APPROVED** [review]
👍

**Comment by larshp** [comment]
thanks 👍

**Pull request merged**